### DISCUSSION TITLE
Where can i get file uri while restoring from snapshot in Qdrant Cloud.

### DISCUSSION THREAD

**Discussion by chapimenge3** [qdrant/qdrant#3747] [Q&A] [answered]
The documentation of Qdrant mentions about restoring from file uri inside of qdrant cloud. 

**IN SHORT:** i want to recover snapshot in qdrant cloud using the api with out downloading and uploading it again.(possibly with FILE URI)

> When recovering from a URL, the URL must be reachable by the Qdrant node that you are restoring. In Qdrant Cloud, restoring via URL is not supported since all outbound traffic is blocked for security purposes. **You may still restore via file URI or via an uploaded file.**

https://qdrant.tech/documentation/concepts/snapshots/#recover-from-a-url-or-local-file

But i can't find anywhere mentioning how i can build the file UI from the qdrant cloud. 

File URI can be `file:///tmp/snapshot-2022-10-10.snapshot` but in case of qdrant cloud where does this file path can be found?

I don't want to download and restore from file if possible since the snapshot is already inside of Qdrant cloud but i can't seem to find any way to get the file path or file URI that i can give to the below request

```bash
PUT /collections/{collection_name}/snapshots/recover?wait=false
{
  "location": "https://QDRANT_HOST/collections/COLLECTION_NAME/snapshots/SNAPSHOT_NAME.snapshot"
}
```

But it is not working like above i tried many combinations but i couldn't achieve the goal. 

Can somebody help?

**IN SHORT** i want to recover snapshot in qdrant cloud using the api with out downloading and uploading it again.(possibly with FILE UI

**Comment by timvisee** [comment]
I'm not sure if we support using file URIs in cloud. I'll forward your question to the cloud team, as they're better qualified to answer this.

**Reply by chapimenge3** [reply]
![image](https://github.com/qdrant/qdrant/assets/48279189/80c365d8-7f4c-46ac-b41e-5df5de64d778)

They still mention the File URI is possible. That is why i asked.

**Comment by mikheillomidzeq** [comment, marked as answer]
Hi @chapimenge3 , 

Qdrant cloud snapshots are saved in default directory which is `/qdrant/snapshots`. Using this info you can constract snapshot file URI and use it. 

Here is the HTTP request template for it:
```http
PUT collections/{collection_name}/snapshots/recover
{
  "location": "file:///qdrant/snapshots/{collection_name}/{snapshot_name}"
}
```

**Reply by chapimenge3** [reply]
Thank you very much @mikheillomidzeq Works like a charm.

**Reply by chapimenge3** [reply]
Would be really helpful if you add this in the docs or i can add it incase Qdrant accept changes on the docs
